core/metrics_manager.js

```diff
@@ -280,40 +231,145 @@ Blockly.MetricsManager.prototype.getViewMetrics = function(
 
 /**
  * Gets content metrics in either pixel or workspace coordinates.
- *
- * This can mean two things:
- * If the workspace has a fixed width and height then the content
- * area is rectangle around all the top bounded elements on the workspace
- * (workspace comments and blocks).
- *
- * If the workspace does not have a fixed width and height then it is the
- * metrics of the area that content can be placed. This area is computed by
- * getting the rectangle around the top bounded elements on the workspace and
- * adding padding to all sides.
+ * The content area is a rectangle around all the top bounded elements on the
+ * workspace (workspace comments and blocks).
  * @param {boolean=} opt_getWorkspaceCoordinates True to get the content metrics
  *     in workspace coordinates, false to get them in pixel coordinates.
- * @param {!Blockly.MetricsManager.ContainerRegion=} opt_viewMetrics The view
- *     metrics if they have been previously computed. Not passing in view
- *     metrics may cause them to be computed again.
  * @return {!Blockly.MetricsManager.ContainerRegion} The
  *     metrics for the content container.
  * @public
  */
 Blockly.MetricsManager.prototype.getContentMetrics = function(
-    opt_getWorkspaceCoordinates, opt_viewMetrics) {
-  var scale = opt_getWorkspaceCoordinates ? this.workspace_.scale : 1;
-  var contentDimensions = null;
-  if (this.workspace_.isContentBounded()) {
-    opt_viewMetrics = opt_viewMetrics || this.getViewMetrics(false);
-    contentDimensions = this.getContentDimensionsBounded_(opt_viewMetrics);
-  } else {
-    contentDimensions = this.getContentDimensionsExact_();
+    opt_getWorkspaceCoordinates) {
+  var scale = opt_getWorkspaceCoordinates ? 1 : this.workspace_.scale;
+
+  // Block bounding box is in workspace coordinates.
+  var blockBox = this.workspace_.getBlocksBoundingBox();
+
+  return {
+    height: (blockBox.bottom - blockBox.top) * scale,
+    width: (blockBox.right - blockBox.left) * scale,
+    top: blockBox.top * scale,
+    left: blockBox.left * scale,
+  };
+};
+
+/**
+ * Returns whether the scroll area has fixed edges.
+ * @return {boolean} Whether the scroll area has fixed edges.
+ * @package
+ */
+Blockly.MetricsManager.prototype.hasScrollEdges = function() {
+  // This exists for optimization of bump logic.
+  return !this.workspace_.isMovableHorizontally() ||
+      !this.workspace_.isMovableVertically();
+};
+
+/**
+ * Computes the fixed edges of the scroll area.
+ * @param {!Blockly.MetricsManager.ContainerRegion=} opt_viewMetrics The view
+ *     metrics if they have been previously computed. Passing in null may cause
+ *     the view metrics to be computed again, if it is needed.
+ * @return {Blockly.MetricsManager.FixedEdges} The fixed edges of the scroll
+ *     area.
+ * @protected
+ */
+Blockly.MetricsManager.prototype.computeFixedEdges_ = function(
```

Comment:
Since we are still getting something maybe `getComputedFixedEdges_`?

Reply:
sgtm